### PR TITLE
feat: [SUB-BRANCH] Create a confirmation modal for leave a channel

### PR DESCRIPTION
**Pull request by sandrade-dcl**
## What does this PR change?
Create a confirmation modal to be showed each time the user want to leave a channel:

image

## How to test the changes?
1. Go to: https://play.decentraland.zone/?renderer-branch=feat/confirmation-modal-for-leave-a-channel
2. Try to leave a channel (from the conversation list, from a specific channel window or from the search channels window).
3. Notice that a new confirmation modal is open and let you confirm or cancel the leave.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs [hunk 1]
```diff
@@ -753,6 +756,20 @@ private void OpenChannelCreation()
...
 
+    public void AddChannelLeave(LeaveChannelWindowController controller)
+    {
+        if (controller.View.Transform.parent == view.fullScreenWindowContainer) return;
+
+        controller.View.Transform.SetParent(view.fullScreenWindowContainer, false);
+        channelLeaveWindow = controller;
+    }
+
+    private void OpenChannelLeave(string channelId)
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs:767]: OpenChannelLeaveConfirmation

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Interfaces/IHUDFactory.cs [hunk 2]
```diff
@@ -64,8 +64,9 @@ public enum HUDElementID
         CHANNELS_CHAT = 34,
         CHANNELS_SEARCH = 35,
         CHANNELS_CREATE = 36,
+        CHANNELS_LEAVE = 37,
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Interfaces/IHUDFactory.cs:67]: CHANNELS_LEAVE_CONFIRMATION

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs [hunk 3]
```diff
@@ -753,6 +756,20 @@ private void OpenChannelCreation()
         channelCreationWindow.SetVisibility(true);
     }
 
+    public void AddChannelLeave(LeaveChannelWindowController controller)
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs:759]: AddChannelLeaveConfirmation

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/ILeaveChannelWindowComponentView.cs [hunk 4]
```diff
@@ -0,0 +1,17 @@
+using System;
+using UnityEngine;
+
+namespace DCL.Chat.HUD
+{
+    public interface ILeaveChannelWindowComponentView
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/ILeaveChannelWindowComponentView.cs:6]: ILeaveChannelConfirmationWindowComponentView

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/LeaveChannelWindowController.cs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        internal ILeaveChannelWindowComponentView joinChannelView;
+        internal IChatController chatController;
+
+        public event Action<string> OnLeaveChannel;
+
+        public LeaveChannelWindowController(IChatController chatController)
+        {
+            this.chatController = chatController;
+
+            this.chatController.OnChannelLeft += HandleChannelLeft;
```
Comment (lorux0) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/LeaveChannelWindowController.cs:16]: perhaps you could subscribe in the `Initialize` method

**Review by lorux0 — APPROVED** [review]

**Comment by RominaMarchetti** [comment]
@sandrade-dcl the pop up doesn't appear when trying to leave a channel from the channel window. 
On a different note, I couldn't test leaving a channel from the channel browser because the search result doesn't show me the channels I belong.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs (hunk 1 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs]: Done!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Interfaces/IHUDFactory.cs (hunk 2 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/Interfaces/IHUDFactory.cs]: Done!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs (hunk 3 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/TaskbarHUD/TaskbarHUDController.cs]: Done!

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/ILeaveChannelWindowComponentView.cs (hunk 4 above)
Comment (sandrade-dcl) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/WorldChatWindowHUD/ILeaveChannelWindowComponentView.cs]: Done!

**Comment by sandrade-dcl** [comment]
> @sandrade-dcl the pop up doesn't appear when trying to leave a channel from the channel window. On a different note, I couldn't test leaving a channel from the channel browser because the search result doesn't show me the channels I belong.

Fixed! :) Thanks Romi!

**Pull request merged**